### PR TITLE
[zeenews] Add extractor

### LINKED ISSUE #4967: [zeenews.india.com] Site support request.
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

India

### Example URLs

https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750

https://zeenews.india.com/hindi/india/video/videsh-superfast-queen-elizabeth-iis-funeral-today/1357710

### Provide a description that is worded well enough to be understood

Hello
Although I have no experience and mastery in programming yt-dlp. But it might be useful for you to know that the blob url is available for the video. for eg: 
blob:https://zeenews.india.com/605a3e30-d354-4358-84d5-3dfdebe6b539

I extracted this from "https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750" this url.

It might be possible to get the video file from this url and can be easily downloaded through yt-dlp.
If this does not work then I apologize. 

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750
[debug] Command-line config: ['-vU', 'https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750']
[debug] User config "C:\Users\user\AppData\Roaming\yt-dlp\config.txt": ['--all-subs', '--embed-subs', '--embed-thumbnail', '--add-metadata', '--write-annotations', '--download-archive=All Youtube History.txt', '--write-annotations', '--write-description', '--write-info-json', '--write-thumbnail', '-f', 'bv+ba/best', '--merge-output-format', 'mp4', '-ciw', '--retries', 'infinite', '-o', '%(title)s-%(id)s-%(format_id)s.%(ext)s']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev (setts), ffprobe 4.4-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Loading archive file 'All Youtube History.txt'
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [generic] Extracting URL: https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750
[generic] 1322750: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 1322750: Extracting information
[debug] Looking for Brightcove embeds
[debug] Looking for embeds
ERROR: Unsupported URL: https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1459, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1535, in __extract_info
  File "yt_dlp\extractor\common.py", line 670, in extract
  File "yt_dlp\extractor\generic.py", line 3078, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://zeenews.india.com/hindi/world/video/the-china-files-how-did-xi-jinping-become-the-most-powerful-and-only-leader-of-chinas-power/1322750
```

### PR DESCRIPTION
**Pull request by m4tu4g**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE

Closes #4967

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/zeenews.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def _real_extract(self, url):
+        content_id = self._match_id(url)
+        webpage = self._download_webpage(url, content_id)
+        info = self._search_json(
+            r'/videos"}]}</script></div><div><script type="application/ld\+json">',
+            webpage,
+            r'</script> </div><div><script type="application/ld\+json">',
+            content_id
+        )
```
Comment (pukkandan) [inline yt_dlp/extractor/zeenews.py:46]: use self._yield_json_ld

File: yt_dlp/extractor/zeenews.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class ZeeNewsIE(InfoExtractor):
+
+    _VALID_URL = r'''(?x)
+                    (?:
+                    zeenews:|
+                    https?://(?:zeenews\.)?india\.com/(?:[^#?]+/)(?:video)/)
+                    (?P<display_id>[^#/?]+)/(?P<id>[^#/?]+)/?(?:$|[?#])
+                    '''
```
Comment (pukkandan) [inline yt_dlp/extractor/zeenews.py:12]:
```suggestion
    _VALID_URL = r'https?://(?:zeenews\.)?india\.com/[^#?]+/video/(?P<display_id>[^#/?]+)/(?P<id>[^#/?]+)/?(?:$|[?#])'
```

1. Use `display_id`
1. Add test case without `zeenews.`
1. Add test case for non-number `id`

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/zeenews.py (hunk 2 above)
Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py:12]: r'https?://zeenews\.india\.com/[^#?]+/video/(?P<display_id>[^#/?]+)/(?P<id>[0-9]+)'

?

File: yt_dlp/extractor/zeenews.py (hunk 1 above)
Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py:46]: need start and end patterns but content_id is not required

Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py]: should be fine now

File: yt_dlp/extractor/zeenews.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from .common import InfoExtractor
+from ..utils import unified_strdate
+
+
+class ZeeNewsIE(InfoExtractor):
+
+    _VALID_URL = r'https?://zeenews\.india\.com/[^#?]+/video/(?P<display_id>[^#/?]+)/(?P<id>[0-9]+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/zeenews.py:7]:
```suggestion
    _VALID_URL = r'https?://zeenews\.india\.com/[^#?]+/video/(?P<display_id>[^#/?]+)/(?P<id>\d+)'
```

File: yt_dlp/extractor/zeenews.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                'display_id': 'videsh-superfast-queen-elizabeth-iis-funeral-today',
+                'upload_date': '20220919',
+                'thumbnail': r're:^https?://.*\.jpg*',
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        match = self._match_valid_url(url).groupdict()
+        content_id, display_id = match['id'], match['display_id']
```
Comment (pukkandan) [inline yt_dlp/extractor/zeenews.py:37]:
```suggestion
        content_id, display_id = self._match_valid_url(url).group('id', 'display_id')
```

File: yt_dlp/extractor/zeenews.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+        return {
+            'id': content_id,
+            'description': info.get('description'),
+            'title': info.get('name'),
+            'display_id': display_id,
+            'formats': formats,
+            'upload_date': unified_strdate(info.get('uploadDate')),
+            'thumbnail': info.get('thumbnailUrl'),
+        }
```
Comment (pukkandan) [inline yt_dlp/extractor/zeenews.py:53]:
```suggestion
        json_ld_list = list(self._yield_json_ld(webpage, display_id))

        embed_url = traverse_obj(json_ld_list, (lambda _, v: v['@type'] == 'VideoObject', 'embedURL'))
        formats = self._extract_m3u8_formats(embed_url, content_id, 'mp4')
        self._sort_formats(formats)

        return {
            **self._json_ld(json_ld_list, display_id),
            'id': content_id,
            'display_id': display_id,
            'formats': formats,
        }
```
does this work?

Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py:51]: can't traverse correctly

Comment (Lesmiscore) [reply yt_dlp/extractor/zeenews.py:51]: Add `get_all=False` to the new line 39
Then it should work

Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py:51]: `get_all=False` fixed traverse issue

Comment (m4tu4g) [reply yt_dlp/extractor/zeenews.py:51]: but there seems to be problem with L44, where `title` field is not being sent

**Pull request merged**